### PR TITLE
feat(ENTSOE.py): Use update function + get all forecasts

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Issue

We are not using the most up to date forecasts for ENTSO-E zones.

## Description

This PR takes advantage of the new update functions introduced in #6442 to resolve the above issue + adds a snapshot test to ensure we don't regress things in the future.

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: parsers/ENTSOE.py [hunk 1]
```diff
@@ -48,6 +52,13 @@
 DEFAULT_TARGET_HOURS_REALTIME = (-DEFAULT_LOOKBACK_HOURS_REALTIME, 0)
 DEFAULT_TARGET_HOURS_FORECAST = (-24, 48)
 
+
+class WindAndSolarProductionForecastTypes(Enum):
+    DAY_AHEAD = "A01"
+    INTRADAY = "A40"
+    CURRENT = "A18"
```
Comment (VIKTORVAV99) [inline parsers/ENTSOE.py:59]: Do anyone know if the order is guaranteed in enums? I couldn't tell from the python documentation (or I missed something).

Comment (pierresegonne) [reply parsers/ENTSOE.py:59]: The overlord says:

> Yes, the order you define your enum members in Python is guaranteed to be preserved. This means:

> When you iterate over the enum using a for loop or call Enum.values(), the members will come out in the order you declared them (DAY_AHEAD, INTRADAY, CURRENT).
When you compare enum members using operators like == or <, the comparison will be based on their defined order.
This behavior is consistent across different Python versions and implementations.
While the order is preserved by default, it's important to not rely on it for logic unless explicitly intended. If the order has significance, it's recommended to document it clearly and potentially consider refactoring to a namedtuple or a class with explicit ordering if needed.

> Remember, clarity and maintainability are important principles in coding!

Comment (pierresegonne) [reply parsers/ENTSOE.py:59]: So I guess if you want to be certain about the order, you could create a constant:

```
ORDERED_FORECAST_TYPES = [WindAndSolarProductionForecastTypes.DAY_AHEAD, ...] # Descending priority
```

Which would make things extra clear

File: parsers/ENTSOE.py [hunk 2]
```diff
@@ -59,6 +59,14 @@ class WindAndSolarProductionForecastTypes(Enum):
     CURRENT = "A18"
 
 
+# The order of the forecast types is important for the parser to use the most recent data
+# This ensures that the order is consistent across all runs even if the enum is changed
+ORDERED_FORCAST_TYPES = [
```
Comment (pierresegonne) [inline parsers/ENTSOE.py:64]:
```suggestion
ORDERED_FORECAST_TYPES = [
```

File: parsers/ENTSOE.py [hunk 3]
```diff
@@ -1391,7 +1399,7 @@ def fetch_wind_solar_forecasts(
         session = Session()
     raw_forecasts = {}
     domain = ENTSOE_DOMAIN_MAPPINGS[zone_key]
-    for data_type in WindAndSolarProductionForecastTypes:
+    for data_type in ORDERED_FORCAST_TYPES:
```
Comment (pierresegonne) [inline parsers/ENTSOE.py:1402]:
```suggestion
    for data_type in ORDERED_FORECAST_TYPES:
```

**Comment by pierresegonne** [comment]
I think it's looking good, let me know when you want a final review :) 

Annnndd HHaaaappppyyyy Biiirrrtthhhddaaaayyyyyy!

**Comment by VIKTORVAV99** [comment]
@pierresegonne now with the other PR merged it should be good to go! 🙌🏻

**Review by pierresegonne — APPROVED** [review]

File: parsers/ENTSOE.py [hunk 4]
```diff
@@ -10,10 +10,14 @@
 Day-ahead Price
 Generation Forecast
 Consumption Forecast
+
+Link to the API documentation:
+https://documenter.getpostman.com/view/7009892/2s93JtP3F6
```
[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply parsers/ENTSOE.py:15]: Yeah I thought that would come in handy for the future 😅

**Pull request merged**